Introduce Irmin stats within Tezos

maiste opened the pull request:
This PR is a draft for exposing the Irmin statistics from the `irmin-pack` library within Tezos through the Prometheus system.
It is a continuation of mirage/irmin#1574.

To be able to test this feature, you need to:
- Have Prometheus running on your machine
  ```
  sudo systemctl start prometheus
  ```
- Ensure that you have copied this text at the end of `/etc/prometheus/prometheus.yml`:
   ```yml
    - job_name: 'netdata-scrape'
      metrics_path: "/api/v1/allmetrics"
      params:
        format: [prometheus]
      static_configs:
        - targets: ['localhost:19999']
    - job_name: 'tezos-exporter'
      metrics_path: "/metrics"
      static_configs:
        - targets: ['localhost:9091']
    ```
- Run a tezos node with `--metrics-addr="localhost:9091"`

Commit:
Ppx_expect: change ppx_inline_test -> ppx_expect

Commit:
Ppx_expect: make proto_alpha use ppx_expect

Commit:
Ppx_expect: expect_test in lib_micheline

Commit:
Ppx_expect: lib_stdlib uses ppx_expect

Commit:
Ppx_expect: convert bloomer.ml to expect_test

Commit:
Doc: document ppx_expect

Commit:
Lib: add new expect-test-helper

Commit:
Bench: add check to encoding bench helpers

taiseiKMC committed:
doc: Add doc about merkle-proof encoding formats

mbouaziz committed:
Proto/plugin: view errors make sense for originated contracts only

Commit:
Manifest: rename tezt into tezt_lib

Commit:
lib_context: move disk impl of context to tezos-context.disk

Commit:
lib_context: split context_dump off tezos-context.disk

Commit:
lib_context: factor unix-free module type away for future reuse

solled committed:
doc: added src links

Commit:
lib_context: let in-memory context implement TEZOS_CONTEXT

Commit:
lib_protocol_env,lib_delegate: factor context_ops in lib_proto_env

Commit:
lib_proto_env: add operations to context_ops

Commit:
lib_validation: work on Environment_context.Context.t only

Commit:
lib_shell,lib_store,bin_validation: propagate use of abstract ctxt

etomknudsen committed:
Store: doc: fix typo in output

mbouaziz committed:
Proto/RPCs: contract_big_map_get accepts only originated contracts

mbouaziz committed:
Proto/RPCs: contract_single_sapling_get_diff accepts only originated contracts

mbouaziz committed:
Proto/RPCs: list_entrypoints accepts only originated contracts

mbouaziz committed:
Proto/RPCs: entrypoint_type accepts only originated contracts

Commit:
bin_node: add rpc request middleware to collect metrics

mbouaziz committed:
Proto/Client: get_script/get_storage work on originated contracts only

mbouaziz committed:
Proto: Contract.get_script work for originated contracts only

mbouaziz committed:
Proto: inline origination record in internal origination

mbouaziz committed:
Proto: inline origination record in external origination

mbouaziz committed:
Proto: code and unparsed storage of internal origination do not need to be lazy

Commit:
manifest: remove superflous synopsis, fix typo

mbouaziz committed:
Plugin: get_script work on originated contracts only

mbouaziz committed:
Proto/Michelson: Runtime_contract_error happens only on originated contracts

mbouaziz committed:
Proto/Contract_services: generalize register_field helpers

mbouaziz committed:
Proto/RPCs: Contract.script work on originated contracts only

mbouaziz committed:
Proto/RPCs: Contract.script work on originated contracts only

mbouaziz committed:
Proto/RPCs: simplify register_originated_opt_field

Commit:
lib_context: functorize test_context, run for disk and memory impl

mbouaziz committed:
Proto/RPCs: Contract.storage work on originated contracts only

mbouaziz committed:
Proto/RPCs: Contract.storage work on originated contracts only

mbouaziz committed:
Proto/RPCs: Contract.info get script only for originated contracts

NicNomadic committed:
doc: fix command for snapshot import in howtouse.rst

Commit:
Manifest: be able to specify runtest alias

Commit:
Manifest: be able to declare local Tezt tests

taiseiKMC committed:
doc: Modify description about merkle-proof

taiseiKMC committed:
doc: Make subsubheadings lowercase in merkle-proof-formats

NicNomadic committed:
doc: fix address of faucets

ACoquereau committed:
Proto/apply: do not reuse gas consumed by precheck in apply

Co-authored-by: Diane Gallois-Wong <diane.gallois-wong@nomadic-labs.com>

[Precheck] no longer returns the gas it consumed. Instead, [apply]
should consume all the gas again: that's why we now set
[consume_deserialization_gas] to [Always] when applying an external
manager operation. Similarly, when applying a [Tx_rollup_submit_batch],
[hash_cost message_size] consumed in [precheck] is now also consumed in
[apply].

This refactoring makes [precheck] and [apply] less interdependent, as
a first step toward making [precheck] into a [validate_operation]
function that covers any kind of operation and is called directly
from [Main] (cf %Pipelining).

mbouaziz committed:
Proto/Michelson: implement CONTRACT for SCORU addresses

Fixes #2800

maiste committed:
lib_context: introduce Prometheus

Introduce the `context_metrics` module to
keep track of metrics in Irmin.

sribst committed:
proto/scoru: carbonation cost fct for rmp message

NicNomadic committed:
doc: explain RPC prefixes at the beginning of the RPC reference pages

jobjo committed:
Test: add test for zero-amount-ticket

NicNomadic committed:
doc: mention that manager_key rpc is only for implicit contracts (#2876)

Ngoguey42 commented on src/lib_context/context.ml:
```diff
@@ -675,6 +675,7 @@ module Make (Encoding : module type of Tezos_context_encoding.Context) = struct
            ~lru_size:!lru_size
            root)
     in
+    Context_metrics.collect () ;
```
I'm not fan of having this called from `init`. Could this be placed at topelevel? Or could `Context_metrics.collect` be called from elsewhere in tezose?

Ngoguey42 replied:
Did you mean `if not readonly`?

Ngoguey42 replied:
Tezos will only have one RW instance open at one time. But nothing prevents Tezos to open/close/open. Could you explicitly prevent multiple calls to `collect`?

maiste replied:
Done.

Ngoguey42 commented on src/lib_context/context_metrics.ml:
```diff
@@ -0,0 +1,257 @@
+(* FIXME: https://github.com/tarides/tezos/issues/20
+    When moving to the new Irmin version, the stats from
+    Irmin_pack_unix.Stats.Index should be insert bellow. *)
+module Index = struct end
```
`Index.Stats.get` can get you the stats meanwhile

maiste replied:
Done.

Ngoguey42 reviewed (commented):
Thanks

eugenz-nl committed:
docs: fix typo

maiste committed:
lib_context: support Index stats

Ngoguey42 commented:
Looks good

Commit:
Proto: consume constant cost of manager operation in precheck

Co-authored-by: Albin Coquereau <albin.coquereau@nomadic-labs.com>

This gas consumption should have been part of precheck from the beginning.

This makes some operations that used to fail in apply for lack of gas, to
fail in precheck instead (with the same Operation_quota_exceeded error).

Update some tests accordingly:
- In test_reveal.ml, test_no_reveal_when_exhausted was expected to fail
  in apply for lack of gas, so we replace expect_apply_failure with
  expect_failure.
- In test_delegation.ml, test_self_delegation_emptying_contract is
  expected to fail in apply with another error, but our changes
  caused it to fail in precheck because of the gas, so we increased its
  gas_limit.
- In test_gas_levels.ml, many tests used to fail in apply for lack of
  gas, but it wasn't important since the block gas level is updated
  during precheck. Having them fail in precheck instead is a problem,
  so we increased the gas_limit of the relevant operations.

More test updates to come in the next commits

Commit:
Manifest: fix comment

Commit:
Tezt: expose Base.project_root

Commit:
Alpha/test: migrate regression tests to manifest+tezt

Commit:
Scripts: fix snapshot_alpha.sh for files with spaces

NicNomadic committed:
doc: fix broken links reported by make linkcheck

picdc-work committed:
WASM: add `to_string_unstable` function to Chunked_byte_vector

Commit:
Tezt: add ?file to Regression.register

NicNomadic committed:
doc: replace apt by apt-get in compile-sources.sh (former is discouraged in scripts)

NicNomadic committed:
doc: fix intro to test networks in index.rst

mbouaziz committed:
Proto/Scoru: parameters_type returns an option

mbouaziz committed:
Proto/Sc_rollup: expose chosen errors

Commit:
Build: cleanup static build

taiseiKMC committed:
doc: Remove duplicated sections in merkle-proof-formats tree2

Commit:
Opam: bump

Commit:
Proto,SCORU: Extend PVM interface with proofs about outbox

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

Commit:
Proto,SCORU: Extend PVM interface with proofs about outbox

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

Commit:
Proto,SCORU: Adapt refutation game proofs to new PVM interface

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

mbouaziz committed:
Proto/Ticket_operations_diff: split Transaction_to_contract case

mbouaziz committed:
Proto/Ticket_operations_diff: no need to check entrypoint type

Fix #2351, #2653

mbouaziz committed:
Proto/Ticket_operations_diff: simplify tickets_of_transaction

thomasathorne committed:
Proto: SCORU: add dissection distribution check

picdc-work committed:
WASM/Parser: incremental string parsing to chunked byte vector

jtcoolen committed:
p2p: Fix signature of Metadata.write in p2p_socket

picdc-work committed:
WASM/Parser: UTF8 name small step

Commit:
CI: Use dependency resto.0.8

Commit:
opam: Bump resto to 0.8

picdc-work committed:
WASM/Parser: import/export small step

raphael-proust committed:
Proto-env: remove dependency from structs to s_packer

Use vanilla dune files to assemble the structs backward compatibility
module.

thomasathorne committed:
Proto: SCORU: Add unit test for badly destributed dissections

mbouaziz committed:
Tests/Python: update gas in regression traces

Commit:
Proto,SCORU: Expose SCORU outbox message encoding

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

Commit:
Proto,SCORU: Make the Arith PVM output

Commit:
Proto/apply: remove prepare_apply_manager_operation_content

Various changes have resulted in [prepare_apply_manager_operation_content]
factorizing very little code, and making the beginning of
[apply_external_manager_operation_content] and
[apply_internal_manager_operation_content] needlessly complicated.
We remove it and inline its content instead.

Commit:
Proto: do not decode scripts in precheck

Precheck no longer tries to deserialize scripts. It does still
check that the operation has enough gas for these deserializations
(by consuming an estimated gas cost based on the bytes size: this
has not changed).

This is a semantic change: if the deserialization fails, the
operation will now pass precheck but its application will fail.
This means that the operation may be included in a block, in
which case the fees will be taken but the operation will have
no further effects.

NicNomadic committed:
add dependency of linkcheck and redirectcheck on html and enable parallelism

Commit:
Manifest: require qcheck-alcotest >= 0.18

Commit:
Manifest: require prometheus >= 0.5

Commit:
Manifest: tezos-stdlib and tezos-test-helpers require OCaml >= 4.12

NicNomadic committed:
doc: fix CI job linkcheck and remove dependency to build_all

Commit:
Manifest: add ?opam_with_test

Commit:
Manifest: disable opam tests for sapling

picdc-work committed:
WASM/Parser: centralize exceptions

Commit:
Manifest: tezos-lwt-result-stdlib: only run tests on 64bit

lthms committed:
Sc_rollup,tests: Instantiate and test the WASM PVM

Commit:
SCORU: WASM: Support monadic composition in Lazy_map and Chunked_byte_vector

Commit:
SCORU: WASM: Introduce Lwt into the reference interpreter

picdc-work committed:
WASM/Parser: implement Buffer using Chunked byte vector

jobjo committed:
Test/Michelson: test parse-contract-data for sc rollups

Commit:
Proto,SCORU: Add tests for arithmetic PVM output proofs

We had to make the PVM capable of producing proofs. Thanks
to Thomas Letan who provided us with the correct Irmin
context spell.

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

NicNomadic committed:
doc: move CI job linkcheck to stage "doc" (hence, also in the scheduled pipeline)

jobjo committed:
Proto,SCORU: Improve code quality

Commit:
Proto,SCORU: Document the output facility of arith PVM

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

Commit:
Proto,SCORU: Use error, not string, to represent errors

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

iguerNL committed:
Proto/Client: move param from/to Client_proto(_context_commands/_args)

Commit:
Proto/bench: add check to interpreter bench helpers

Commit:
Proto/bench: add ADX check to BLS interpreter benchmarks

ACoquereau committed:
Tezt: adapt tests with manager_opearation_gas_cost

Commit:
Proto/bench: add ADX check to BLS encoding benchmarks

NicNomadic committed:
doc: remove top label in multinetwork.rst

lthms committed:
Env,proto: Expose a dummy WASM [step] function to the protocol

In this patch, we enrich the V6 protocol with a functor to derive a
WASM step function.

The functor is parameterized with an Irmin tree, so that it can easily
be used with different context implementation.  This is necessary as
long as (1) the functor of the WASM PVM is parameterized by a tree,
and (2) the protocol only knows about one Irmin tree (32-arity,
specific to Tezos needs).

iguerNL committed:
Proto/Client: normalize arguments descriptions

Commit:
Proto/apply: remove identity let binding

Remove "let before_operation = ctxt". Instead, use directly
the function argument, and give a clearer name.

Commit:
Proto/Tests: change gas limit argument and adapt tests

vch9 committed:
Scoru,Proto: avoid [consume_n_messages] unnecessary transfo to int for [n]

vch9 committed:
Scoru,Proto: use [fail_when] when possible

vch9 committed:
Scoru,Proto: clean up a bit the comments

vch9 committed:
Lib_p2p: use [Internal_for_tests] instead of hiding values

Commit:
Proto/Snoop: update ticket benchmarks and register in snoop

Commit:
Proto/Tests: add a test checking that emptying an account costs gas

Commit:
Snoop: Remove Obj.magic

Commit:
Proto/Snoop: update timestamp benchmark with sqrt model

Commit:
Manifest: require uri >= 2.2.0

raphael-proust committed:
Proto-env: flatter file structure in structs

NicNomadic committed:
doc: rephrasing in mockup.rst

mbouaziz committed:
Proto/Apply_results: remove consumed_gas field in encodings

taiseiKMC committed:
doc: Remove description about tag for length in merkle-proof-format/v2-tree[2|32]

Commit:
Merge branch 'rpc_metrics' into 'master'

Add a metric family to count the time spent by rpc call and the number of their occurrences

See merge request tezos/tezos!5240

raphael-proust committed:
Proto-env: merge two structs files always used in conjunction

gh-naih committed:
Proto/SCORU/Test: add a third staker.

gh-naih committed:
Proto/SCORU/Test: initialization with three stakers.

By copy/pasting and adapting originate_rollup_and_deposit_with_two_stakers.

gh-naih committed:
Proto/SCORU/Test: extract dissection initialization.

Because we will use it too.

gh-naih committed:
Proto/SCORU/Test: simplify init_dissection.

gh-naih committed:
Proto/SCORU/Test: extract refutation initialization.

Because we will use it too.

gh-naih committed:
Proto/SCORU/Test: staker's injectivity.

Commit:
Merge branch 'mehdi@ticket_op_diff' into 'master'

Proto/Ticket_operations_diff: no need to check entrypoint type

Closes #2653 and #2351

See merge request tezos/tezos!5494

victor-dumitrescu committed:
lib_crypto: upgrade bls12-381 to 4.0.0

thomasathorne committed:
Proto: SCORU: relax dissection rules slightly

Co-authored-by: Yann Regis-Gianas <yann@nomadic-labs.com>

Commit:
Merge branch 'raphael-p@no-s-packer-in-proto-env-struct--issue-1602' into 'master'

Proto-env: remove dependency from structs to s_packer

Closes #1602

See merge request tezos/tezos!5503

jobjo committed:
Proto: rename cost function

jobjo committed:
Proto: pay gas for serializing internal messages

Commit:
Merge branch 'julien.coolen@p2p-socket-fix-arguments-order' into 'master'

p2p: Fix signature of Metadata.write in p2p_socket

See merge request tezos/tezos!5500

Commit:
Merge branch 'sylvain@scoru@carbonated_rmp_messages' into 'master'

scoru: add carbonated function cost for serializing internal inbox message and deserializing outbox message

Closes #2951

See merge request tezos/tezos!5443

thomasathorne committed:
Proto: SCORU: changes to scoru PBTs to reflect dissection rule change

Commit:
Merge branch 'lthms@minimal-wasm-proof' into 'master'

Env,proto: Expose a dummy WASM `step` function to the protocol

See merge request tezos/tezos!5529

Commit:
Merge branch 'scoru@wasm-lwt' into 'master'

SCORU: WASM: Lwt in reference interpreter

See merge request tezos/tezos!5511

Commit:
Merge branch 'lrand@bench-bls-check-portable' into 'master'

Proto/bench: add ADX check to BLS benchmarks

See merge request tezos/tezos!5523

mbouaziz committed:
Proto,SCORU: Improve code quality.

Commit:
Proto,SCORU: Improve code quality

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

Commit:
Proto,SCORU: Improve code quality

Signed-off-by: Yann Regis-Gianas <yann@nomadic-labs.com>

Commit:
Merge branch 'pipelining@do-not-decode-in-precheck' into 'master'

Proto: do not decode scripts in precheck and do not reuse precheck gas in apply

See merge request tezos/tezos!5506

Commit:
Tests/CI: Only pytest alpha

Commit:
Merge branch 'lrand@update-bench-J' into 'master'

QOL updates for some gas benchmarks

See merge request tezos/tezos!5533

jobjo committed:
Proto: missing renames

jobjo committed:
Proto: missing case for execute-outbox-message

Commit:
Tezt: ensure toru manager ops tezts run from Jakarta

[transfer_ticket_deserialization_canary protocols] and
[transfer_ticket_deserialization_too_large protocols] where set up to
support only Alpha instead of any protocol from Jakarta.

mbouaziz committed:
Proto/changelog: add entry

Commit:
Merge branch 'thomas.athorne@scoru-refutation-relax-dissection-rule' into 'master'

Proto: SCORU: relax dissection rules slightly

See merge request tezos/tezos!5543

Commit:
SCORU: WASM: Add tree decoding module

jobjo committed:
Test,SCORU: amend WASM test to use error

Commit:
Merge branch 'rename-sc-atomic-batch-left-over' into 'master'

Proto: SCORU: Rename sc atomic batch left over

See merge request tezos/tezos!5548

Commit:
Merge branch 'mehdi@sc_rollup_CONTRACT' into 'master'

Proto/Michelson: implement CONTRACT for SCORU addresses

Closes #2800

See merge request tezos/tezos!5399

Commit:
SCORU: WASM: Split lazy map and vector

mbouaziz committed:
Proto/Changelog: add entry

thomasathorne committed:
Proto: SCORU: fix bug with conflict point in `init_game`

- the bug was added during refactoring

Commit:
Merge branch 'mehdi@rpcs_contract_hash' into 'master'

Proto/RPCs: restrict some RPCs to originated contracts only

See merge request tezos/tezos!5373

Commit:
Merge branch 'scoru@add-outbox-support-in-PVM' into 'master'

Extend PVM interface with proofs about outbox

Closes #2927

See merge request tezos/tezos!5478

jobjo committed:
Proto: allow-zero flag for ticket-scanner

jobjo committed:
Proto: use ticket-scanner error

jobjo committed:
Proto: amend tests

jobjo committed:
Proto: allow-zero flag for ticket op diffs

jobjo committed:
Test: ticket-scanner test for zero amount

jobjo committed:
Proto: output-proof encoding

jobjo committed:
Proto: use outbox-message

Commit:
Merge branch 'hsaito@merkle-format-doc' into 'master'

Add docs about merkle-proof encoding formats

See merge request tezos/tezos!4936

gh-naih committed:
Proto/SCORU/Test: add new helper functions.

gh-naih committed:
Proto/SCORU/Test: (partly) refactor new_context.

gh-naih committed:
Proto/SCORU/Test: use new helper functions in a few places.

And the rest will be done as part of issue #3195.

vch9 committed:
Proto,Scoru: use a record for [starting_point] in inbox_repr

gh-naih committed:
Proto/SCORU/Test: parameterize init_dissection with init_tick.

Commit:
Manifest: add uri dependency (outside protocols)

Commit:
Manifest: add uri dependency (in protocols)

vch9 committed:
Scoru,Proto: limit the size of a wrapped proof

jobjo committed:
Proto: add outbox-level

jobjo committed:
Test: amend tests with outbox level

Commit:
Merge branch 'pierrick@wasm-parser-shallow-data' into 'master'

WASM / Parser : use Chunked byte vector for data segment

Closes #3165

See merge request tezos/tezos!5495

ferilata committed:
SCORU: Wasm: Add lib_scoru_wasm and call from Proto Env

Commit:
Merge branch 'pierrick@wasm-parser-more-small-step' into 'master'

WASM/Parser: small step UTF8 parsing

Closes #3174

See merge request tezos/tezos!5502

Commit:
Merge branch 'carbonate-internal-message-serialization' into 'master'

Proto, SCORU: Carbonate internal message serialization

See merge request tezos/tezos!5551

Commit:
Merge branch 'thomas.athorne@scoru-dissection-distribution' into 'master'

Proto: SCORU: add dissection distribution check

Closes #2998

See merge request tezos/tezos!5493

jobjo committed:
Test: amend refutation game tests

Commit:
Merge branch 'functori@scoru@refactor-client-args' into 'master'

SCORU: Move CLI args to Client_proto_args module

Closes #3064

See merge request tezos/tezos!5522

Commit:
Merge branch 'vch9@lib_p2p-use-internal-for-tests' into 'master'

Lib_p2p: use [Internal_for_tests] instead of hiding values

See merge request tezos/tezos!5531

Commit:
Merge branch 'vch9@scoru@clean-up-sc-inbox' into 'master'

Scoru,Proto: few clean up for the Inbox module

See merge request tezos/tezos!5530

Commit:
Merge branch 'germanD@toru-manager-tezts-from-Jakarta' into 'master'

Tezt: nit to ensure toru  deserialization manager ops tezts run from Jakarta

See merge request tezos/tezos!5560

Commit:
Merge branch 'jobjo@outbox-level-in-proof' into 'master'

Proto: SCORU: extend output proofs with oubox-level

See merge request tezos/tezos!5559

Commit:
Merge branch 'allow-zero-tickets-flag' into 'master'

Proto: Introduce an allow-zero-amount-tickets flag

See merge request tezos/tezos!5519

Commit:
CI/coverage: make HTML report in collapsible tree format

Commit:
CI/coverage: store coverage traces

NicNomadic committed:
doc: remove makefile target sanity:

Commit:
Merge branch 'hans@scoru-wasm-env' into 'master'

SCORU: Wasm: Add lib_scoru_wasm and call from Proto Env

Closes #3095 and #3090

See merge request tezos/tezos!5482

Commit:
Merge branch 'mehdi@consumed_gas' into 'master'

Proto/Apply_results: remove consumed_gas field in encodings

See merge request tezos/tezos!5536

Commit:
Merge branch 'scoru@wasm-lazy-vector' into 'master'

SCORU: WASM: Split lazy map and vector

See merge request tezos/tezos!5558

Commit:
manifest: fix manifest

Commit:
Proto/Alpha: do not delete implicit contracts unused empty storages

Commit:
Proto/Test: adapt gas test to the fixed semantics

Commit:
Merge branch 'arvid@store_coverage_trace' into 'master'

Tests/Coverage: Store coverage trace and render coverage html in tree form

See merge request tezos/tezos!5542

Commit:
Merge branch 'igarnier-generalize-mem-ctxt' into 'master'

Let lib_validation work on in-memory contexts

See merge request tezos/tezos!5209

vch9 committed:
Proto/SCORU: fix encoding for staker_in_game error.

Commit:
Merge branch 'romain-manifest-backports' into 'master'

Fix opam files (backports from the 13.0 opam release)

See merge request tezos/tezos!5509

Commit:
CI/Coverage: Make coverage.py's unzip non-interactive

ACoquereau committed:
Proto/Test: bounding of gas consumption when emptying implicit

undelegated contract

Commit:
Merge branch 'arvid@ci-master-overwrite-coverage-report' into 'master'

CI/Coverage: Make coverage.py's unzip non-interactive

See merge request tezos/tezos!5570

gh-naih committed:
Proto/SCORU: fix staker in the staker_in_game error.

Commit:
Snoop: avoid 'nologscale' option in incompatible gnuplot versions

Commit:
Merge branch 'scoru@wasm-tree-decoding' into 'master'

SCORU WASM tree decoder

See merge request tezos/tezos!5550

vect0r-vicall committed:
Proto/Changelog: do not delete empty storage

Commit:
Merge branch 'romain-tezt-manifest' into 'master'

Manifest + Tezt = easy to define local Tezt tests

See merge request tezos/tezos!5148

Commit:
Merge branch 'naih@refutation-game-injective-staker' into 'master'

Proto/SCORU: test staker's injectivity in refutation games

Closes #2452

See merge request tezos/tezos!5537

Commit:
CI: fix Tezos Docker image labels (OCI standard)

Commit:
Merge branch 'davdumas@static-layers-1' into 'master'

CI: fix Tezos Docker image labels (OCI standard)

See merge request tezos/tezos!5402

Commit:
Everywhere: fix warning 68

Commit:
Proto/apply: do not hash again in record_operation

Commit:
Build: explicit warnings for the codebase

Commit:
Build: fix -o3 flags

Commit:
Merge branch 'lrand@fix-gnuplot-old-version' into 'master'

Snoop: avoid 'nologscale' option in incompatible gnuplot versions

See merge request tezos/tezos!5575

Commit:
Merge branch 'vch9@scoru@check-size-refutation-proof' into 'master'

Scoru,Proto: limit the size of a wrapped proof

Closes #3062

See merge request tezos/tezos!5555

Commit:
Merge branch 'vbot@do_not_delete_empty_storage' into 'master'

Do not delete empty storage

Closes #3188

See merge request tezos/tezos!5566

victor-dumitrescu committed:
Merge branch 'vdum@bls12-381' into 'master'

Upgrade bls12-381 to 4.0.0

See merge request tezos/tezos!5541

Commit:
Snoop: use base name for plots

Commit:
Merge branch 'diane@do-not-hash-in-record-operation' into 'master'

Proto/apply: do not hash again in record_operation

See merge request tezos/tezos!5582

Commit:
Manifest: improve flags layout

Commit:
Manifest: generate

acerone-trilitech committed:
Scoru: Rollup node: distinguish between local and global endpoints

Commit:
Merge branch 'hhugo-warn' into 'master'

Enable more warnings in the codebase.

See merge request tezos/tezos!5581

Commit:
Merge branch 'hhugo-fix-flags' into 'master'

Build: fix -o3 flags

See merge request tezos/tezos!5580

Commit:
Merge branch 'andrea@scoru-local-global-endpoints' into 'master'

Scoru: Rollup node: distinguish between local and global endpoints

Closes #3001

See merge request tezos/tezos!5512

Commit:
Merge branch 'lrand@snoop-report-graphics-relative-filename' into 'master'

Snoop: use base name for plots

See merge request tezos/tezos!5589

Commit:
Merge branch 'hhugo@static-dune3' into 'master'

Cleanup compilation of static binaries

See merge request tezos/tezos!3978

Commit:
Doc: more examples for expect_test

Commit:
Merge branch 'arvid@only-pytest-alpha' into 'master'

CI/Tests: Only pytest alpha

See merge request tezos/tezos!5546

Commit:
Merge branch 'typo-doc' into 'master'

doc: The typo train

Closes #2876

See merge request tezos/tezos!5244

Commit:
Merge branch 'mehdi@contract_get_script' into 'master'

Proto: Contract.get_script work for originated contracts only

See merge request tezos/tezos!5553

maiste commented:
MR opens on Tezos here: https://gitlab.com/tezos/tezos/-/merge_requests/5600

maiste commented:
As it was closed on `GitLab` I close it here.
